readthedocs/api/v2/views/model_views.py

```diff
@@ -272,6 +272,25 @@ def retrieve(self, *args, **kwargs):
                     )
         return Response(data)
 
+    @decorators.action(
+        detail=True,
+        permission_classes=[permissions.IsAdminUser],
+        methods=['post'],
+    )
+    def restart(self, request, **kwargs):
+        """
+        Restart the build so it can be re-used when re-trying.
+
+        Dates and states are usually overriden by the build,
+        we care more about deleting the commands.
+        """
+        instance = self.get_object()
+        instance.output = ''
+        instance.cold_storage = False
```

Comment:
I think we should reset these also:

- `state`
- `status`
- `success`
- `error`
- `exit_code`
- `builder`

Maybe as well,

- `setup`
- `setup_error`

Reply:
Looks like setup and setup_error aren't used